### ISSUE TITLE
unit test makefile relinking!

### ISSUE THREAD

**Issue by znichola** [sophiakoulen/minishell#23]
The unit test makefile relinks! this is not good.

```make
parsing : all
	$(CC) $(CFLAGS) -I../includes -lreadline -lminishell -L.. -o $@.test $(OBJS_PATH)$@_test.o
```

when `make parsing` is called make can't check if the file `parsing` exists because we compile a file called `parsing.test`
I recommend adding  a varaible `TEST_OUT = .test` then refactoring the unit tests as so:

```make
parsing$(TEST_OUT) : all
	$(CC) $(CFLAGS) -I../includes -lreadline -lminishell -L.. -o $@$(TEST_OUT) $(OBJS_PATH)$@_test.o
```
I'm in the middle of trying to figure out the parsing, so just nothing this down for later!

**Comment by znichola** [comment]
ah no! this won't work because the `@` in the compilation rule would break. Idk what the elegant solution is. we can also just ignore this for now I think.

**Comment by znichola** [comment]
fixed with #155
